### PR TITLE
Jaybee84/move ensemble

### PR DESCRIPTION
**Pull request by jaybee84**
### Purpose

addresses #101 

### Directions for reviewers

Review to see if the flow of the article seems disrupted

#### Which areas should receive a particularly close look?

- gold-standard dataset in intro
- ensemble learning in model-complexity

#### Is there anything that you want to discuss further?

Does this rearrangement hurt the overall message of the article

#### Is the pull request ready for review?

Yes

### Manuscript checklist

Unless otherwise noted above, this PR will be considered ready for review when all four items have been checked.

- [x] All changes to text follow "one sentence per line" [[Manubot instructions](https://github.com/jaybee84/ml-in-rd/blob/draft-branch/USAGE.md#manubot-markdown)]
- [x] All citations follow the [Manubot citation instructions](https://github.com/jaybee84/ml-in-rd/blob/draft-branch/USAGE.md#citations)
- [x] All changes or additions to tables follow the [Manubot table instructions](https://github.com/jaybee84/ml-in-rd/blob/draft-branch/USAGE.md#tables)
- [x] All figures follow the [Manubot figure instructions](https://github.com/jaybee84/ml-in-rd/blob/draft-branch/USAGE.md#figures)
- [x] All terms that need to be added to glossary have been added to #50 
- [x] There are no new spelling errors identified by the Manubot spellchecker

### CODE REVIEW

File: content/04.model-complexity.md [hunk 1]
```diff
@@ -31,4 +31,24 @@ In immune cell signature discovery, elastic-net regression has been used to redu
 A variation of elastic-net, where a two-step regularized logistic regression was used to pre-select an optimal number of genes before implementing elastic-net regularization for gene selection, identified immune cell signatures in an RNA-seq dataset where the number of cells sampled were far fewer than number of genes profiled [@doi: 10.1186/s12859-019-2994-z].  
 Thus, regularization methods like LASSO or elastic-net have been methods of choice where the profiled feature space is substantially larger than the number of samples; these methods should be explored while working with rare disease datasets.
 
-#### Hyperensemble
+#### Ensemble learning
+
+Rare disease datasets not only present few data points but the data points also come with high label-uncertainty thus forming silver standard datasets.
```
Comment (jaclyn-taroni) [inline content/04.model-complexity.md]: It _very well_ may have been my suggestion to kick the gold standard vs. silver standard discussion to the introduction but seeing this makes me think there's a non-zero chance that if someone is not familiar with this concept prior to reading the paper, they will have forgotten what it means by the time they make it to this section.

File: content/04.model-complexity.md [hunk 2]
```diff
@@ -31,4 +31,24 @@ In immune cell signature discovery, elastic-net regression has been used to redu
...
+Implementing machine learning on such data can lead to unstable predictions. 
+In such cases, combining various machine learning methods together (_ensemble learning_) can increase accuracy and stability of the predictions.
+For example, ensemble learning methods like random forests use bagging of independent decision trees that use similar parameters but different paths to form a consensus about the important predictive features hidden in the dataset [@doi:10.1186/1472-6947-13-134, @doi:10.1023/A:1010933404324, @doi:10.1214/aos/1031689014, @doi:10.1177/2045894019890549, @doi:10/btzfh6].
```
Comment (jaclyn-taroni) [inline content/04.model-complexity.md]:
```suggestion
For example, ensemble learning methods like random forests use bagging of independent decision trees that use similar parameters but different paths to form a consensus about the important predictive features hidden in the dataset [@doi:10.1186/1472-6947-13-134; @doi:10.1023/A:1010933404324; @doi:10.1214/aos/1031689014; @doi:10.1177/2045894019890549; @doi:10/btzfh6]. 
```

I believe.

File: content/04.model-complexity.md [hunk 3]
```diff
@@ -31,4 +31,24 @@ In immune cell signature discovery, elastic-net regression has been used to redu
...
+In silver standard datasets, the limited success of the bagging approach has led to the use of ensemble learning or _cascade learning_, where multiple methods leveraging distinct underlying assumptions are used in tandem and augmented with algorithms like AdaBoost (_boosting_), to capture stable patterns existing in the silver standard data and reduce uncertainty [@doi:10.1109/CVPR.2001.990537, @doi:10.1007/978-3-540-75175-5_16, @doi:10.1109/ICPR.2004.1334680].
```
Comment (jaclyn-taroni) [inline content/04.model-complexity.md]:
```suggestion
In silver standard datasets, the limited success of the bagging approach has led to the use of ensemble learning or _cascade learning_, where multiple methods leveraging distinct underlying assumptions are used in tandem and augmented with algorithms like AdaBoost (_boosting_), to capture stable patterns existing in the silver standard data and reduce uncertainty [@doi:10.1109/CVPR.2001.990537; @doi:10.1007/978-3-540-75175-5_16; @doi:10.1109/ICPR.2004.1334680].
```

**Review by jaclyn-taroni — APPROVED** [review]
This move makes sense to me!

File: content/04.model-complexity.md (hunk 1 above)
Comment (jaybee84) [reply content/04.model-complexity.md]:
```suggestion
Rare disease datasets not only present limited data points but, due to inadequate understanding of the diseases, the labels associated with the data points also carry high levels of uncertainty. 
Training ML models using data points with high label-uncertainty (i.e. a silver standard dataset) can lead to unstable predictions.
```

Comment (jaybee84) [reply content/04.model-complexity.md]: rephrased this

File: content/04.model-complexity.md [hunk 4]
```diff
@@ -31,4 +31,24 @@ In immune cell signature discovery, elastic-net regression has been used to redu
 A variation of elastic-net, where a two-step regularized logistic regression was used to pre-select an optimal number of genes before implementing elastic-net regularization for gene selection, identified immune cell signatures in an RNA-seq dataset where the number of cells sampled were far fewer than number of genes profiled [@doi: 10.1186/s12859-019-2994-z].  
 Thus, regularization methods like LASSO or elastic-net have been methods of choice where the profiled feature space is substantially larger than the number of samples; these methods should be explored while working with rare disease datasets.
 
-#### Hyperensemble
+#### Ensemble learning
+
+Rare disease datasets not only present few data points but the data points also come with high label-uncertainty thus forming silver standard datasets.
+Implementing machine learning on such data can lead to unstable predictions.
```
Comment (jaybee84) [inline content/04.model-complexity.md]:
```suggestion
```

**Comment by allaway** [comment]
LGTM, this can be merged, right?

**Comment by jaybee84** [comment]
Ah yes... I thought I had already merged it... oh well, doing it now...

**Pull request merged**